src/redux/users/users.js

```diff
@@ -0,0 +1,70 @@
+import axios from 'axios';
+import { defaultState, initialState } from '../initialState';
+
+const SIGN_IN = 'auth/signin';
+const SIGN_UP = 'auth/signup';
+const SIGN_OUT = 'auth/signout';
+
+const url = 'https://pick-a-doc.herokuapp.com/api/auth/';
+
+export const signIn = (user) => async (dispatch) => {
+  const headers = {
+    'Content-Type': 'application/json',
+  };
+  axios.post(`${url}sign_in`, user, { headers })
+    .then((response) => {
+      initialState.currentUser.isSignedIn = true;
+      initialState.currentUser.attributes = response.data.data;
+      initialState.currentUser.headers = response.headers;
+    })
+    .then(() => {
+      dispatch({
+        type: SIGN_IN,
+        payload: initialState,
+      });
+    });
```

Comment:
This is perfect @eapenzacharias, but why you don't handle if the user for example but invalid credentials? I mean if the request back with an error, not a happy scenario.